libs/d3d12/main.c

```diff
@@ -0,0 +1,392 @@
+/*
+ * Copyright 2018 Józef Kucia for CodeWeavers
+ * Copyright 2020 Joshua Ashton for Valve Software
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ */
+
+#define VK_NO_PROTOTYPES
+#include "vkd3d_win32.h"
+#include "vkd3d_debug.h"
+
+static PFN_vkGetInstanceProcAddr load_vulkan(void)
+{
+    static HMODULE vulkan_module = NULL;
+
+    if (!vulkan_module)
+        vulkan_module = LoadLibraryA("vulkan-1.dll");
```

Comment:
This is not thread-safe.

Reply:
Perhaps just guard it with a static spinlock_t or something trivial.

Reply:
Resolved.